src/views/Spec.vue

```diff
@@ -288,12 +302,85 @@ export default defineComponent({
       }
     })
 
+    const processOauth2Redirect = async () => {
+      const oauth2 = window.opener?.swaggerUIRedirectOauth2
+      const sentState = oauth2.state
+      const redirectUrl = oauth2.redirectUrl
+      let qp
+
+      if (/code|token|error/.test($route.hash)) {
+        qp = $route.hash.substring(1).replace('?', '&')
+      } else {
+        qp = location.search.substring(1)
+      }
+
+      const arr = qp.split('&')
+
+      arr.forEach(function (v, i, _arr) { _arr[i] = '"' + v.replace('=', '":"') + '"' })
+      qp = qp
+        ? JSON.parse('{' + arr.join() + '}',
+          function (key, value) {
+            return key === '' ? value : decodeURIComponent(value)
+          }
+        )
+        : {}
```

Comment:
This algorithm won't be able to handle params that don't have a value (e.g., query params strings like `a=z&x`).

I think it would be better to have an algorithm like
```suggestion
      qp = Object.fromEntries(qp.split('&').map(param => param.split('=')))
```
or if possible use another way of accessing query params on the route as Nate commented.

Reply:
Yeah makes sense, this was a direct lift from the helper functions in swagger-ui. I'll port it over to be more vue-friendly and hopefully catch more edge cases.

Reply:
How does this look, @Mierenga ? I think this should pretty safely account for both cases as well as empty
```ts
// called with $route.hash or $route.query
const parseRouteData = (routeData: string | LocationQuery): Record<string, string> => {
  if (!routeData) return {};

  // handle locationQuery
  if (typeof routeData === 'object') {
    // decode any URI components:
    const routeDataDecoded = Object.keys(routeData).reduce((acc, key) => {
      acc[key] = decodeURIComponent(routeData[key].toString())
      return acc
    }, {})
    return routeDataDecoded
  }

  const data = routeData.startsWith('#') ? routeData.substring(1) : routeData;
  const pairs = data.split('&');
  const result = {};

  pairs.forEach(pair => {
    const [key, value] = pair.split('=');
    result[key] = decodeURIComponent(value);
  });

  return result;
};
```

Meaning that:
```js
window.open('http://localhost:5173/spec/8c655d31-0c89-4660-96d0-829dd92f9e98/oauth2-redirect.html?a=z&x')

const routeHashData = parseRouteData($route.hash);
const routeQueryData = parseRouteData($route.query);
const routeData = { ...routeHashData, ...routeQueryData };

// routeData becomes `{a:'z',x:''}`
```